src/orion/core/worker/experiment.py

```diff
@@ -547,24 +546,23 @@ def _instantiate_config(self, config):
         if self.refers and not isinstance(self.refers.get('adapter'), BaseAdapter):
             self.refers['adapter'] = Adapter.build(self.refers['adapter'])
 
-    def _branch_config(self, conflicts, config):
+    def _branch_config(self, conflicts, branching_configuration):
         """Ask for a different identifier for this experiment. Set :attr:`refers`
         key to previous experiment's name, the one that we branched from.
 
         :param config: Conflicting configuration that will change based on prompt.
         """
-        experiment_brancher = ExperimentBranchBuilder(
-            conflicts, auto_resolution=config.get('auto_resolution'))
+        experiment_brancher = ExperimentBranchBuilder(conflicts, branching_configuration)
 
         if not experiment_brancher.is_resolved or experiment_brancher.auto_resolution:
             branching_prompt = BranchingPrompt(experiment_brancher)
             branching_prompt.cmdloop()
 
-            if branching_prompt.abort:
+            if branching_prompt.abort or not experiment_brancher.is_resolved:
```

Comment:
When is the second condition here expected to happen?

Reply:
Should not, safeguard. Some unit-tests do mock the cmdloop and use this safeguard.